src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/AutoRerouteV2Spec.groovy

```diff
@@ -467,31 +484,23 @@ class AutoRerouteV2Spec extends HealthCheckSpecification {
         }
         isSwitchActivated = true
 
-        then: "Both flows are rerouted"
-        Wrappers.wait(rerouteDelay + WAIT_OFFSET) {
-            withPool {
-                [firstFlow.flowId, secondFlow.flowId].eachParallel { String flowId ->
-                    assert northbound.getFlowStatus(flowId).status == FlowState.UP
-                    assert PathHelper.convert(northbound.getFlowPath(flowId)) != flowPathMap[flowId]
-                }
-            }
-        }
-
-        and: "Flow is rerouted due to switchUp event"
-        //TODO(andriidovhan) specify flow history verification(it is not implemented yet) Reroute reason: switchUp event
+        then: "System tries to reroute the flow on switchUp event"
+        /* there is a risk that flows wont be rerouted, because switch is online but ISL are not discovered yet,
+         that's why we check that system tries to reroute flow on the switchUp event and don't check that flow is UP */
         Wrappers.wait(WAIT_OFFSET / 2) {
-            assert northbound.getFlowHistory(firstFlow.flowId).findAll { it.action == "Flow rerouting" }.size() == 2
-
-            assert northbound.getFlowHistory(secondFlow.flowId).findAll { it.action == "Flow rerouting" }.size() == 5
-            /* 2 = first reroute due to the broken isl + 1 reroute on switch up event;
-             5 = first reroute due to the broken isl + 3 retries + 1 reroute on switch up event;
-             NOTE: retry is available for a flow when switchUp event appears on a transit switch */
+            assert northbound.getFlowHistory(firstFlow.flowId).findAll {
+                it.action == "Flow rerouting"
+            }.last().details == "Reason: Switch '$switchPair1.src.dpId' online"
+            assert northbound.getFlowHistory(secondFlow.flowId).findAll {
+                it.action == "Flow rerouting"
+            }.last().details == "Reason: Switch '$switchPair1.src.dpId' online"
         }
 
```

Comment:
As for me it makes sense to wait for UP status at least as a separate step/wait here before trying to remove the flow